django_netjsongraph/base/link.py

```diff
@@ -103,13 +103,16 @@ def get_from_nodes(cls, source, target, topology):
         :param topology: Topology instance
         :returns: Link object or None
         """
-        source = ';{0};'.format(source)
-        target = ';{0};'.format(target)
         q = (Q(source__addresses__contains=source,
                target__addresses__contains=target) |
              Q(source__addresses__contains=target,
                target__addresses__contains=source))
-        return cls.objects.filter(q).filter(topology=topology).first()
+        objs = cls.objects.filter(q).filter(topology=topology)
+        for obj in objs:
+            if ((source in obj.source.addresses and target in obj.target.addresses) or
+               (target in obj.source.addresses and source in obj.target.addresses)):
+                return obj
+        return None
```

Comment:
please change this to:

```python
source = '"{}"'.format(source)
target = '"{}"'.format(target)
q = (Q(source__addresses__contains=source,
       target__addresses__contains=target) |
     Q(source__addresses__contains=target,
       target__addresses__contains=source))
return cls.objects.filter(q).filter(topology=topology).first()
```

we need to keep the code simple and clear. Since the field uses JSON format, a string will be always delimited by a double quote, so by looking for "10.0.0.1", we are sure we'll only get results containing "10.0.0.1", because 10.0.0.11 will contain "10.0.0.11", which does not contain "10.0.0.1", try:

```
'10.0.0.1' in '10.0.0.11'
True

'"10.0.0.1"' in '"10.0.0.11"'
False
```